### PR TITLE
KOGITO-6883: [SWF Editor] Create a Language Service for Serverless Workflow

### PR DESCRIPTION
**Pull request by tiagobento**
Jira: https://issues.redhat.com/browse/KOGITO-6883

---

On this PR:
- Change `syncpack` to allow for a different version of Monaco to be used on SWF Editor
- Created the new `serverless-workflow-language-service` package.
- Added a new flag to allow for changing the content of an editor without showing the Loading overlay
- Some renames to better represent Diagram VS. Text editors, instead of SVG/Mermaid VS. Monaco.
- Moved all augmentation logic from the SWF Text Editor to the new Language Service package.
- In VS Code, now the default is to open the SWF Editor file using VS Code's built-in text editor. The SWF Diagram Editor is shown side-by-side with the built-in text editor (There's a new configuration option to change that). This behavior was based on the Markdown support for VS Code.
- Changed the `@kie-tools-core/vscode-extension` package to allow for using the CustomTextEditor API as well, where the Multiplying Architecture Editor is synced with the TextDocument on VS Code's backend. This makes multi-tab editing very pleasant and allows for better native support of text editing tools provided by the VS Code platform.
- Added keyboard shortcuts to switch between the built-in text editor in VS Code and our SWF Diagram Editor.

### CODE REVIEW

**Review by paulovmr — APPROVED** [review]

**Comment by tiagobento** [comment]
@tomasdavidorg Can you please test this?

**Comment by tomasdavidorg** [comment]
Hi @tiagobento,
I build the SWF VsCode extension and after I tried to open the diagram panel from a sw.json file I got: 

`command 'extension.kogito.swf.openAsDiagram' not found`

image

File: packages/serverless-workflow-language-service/src/channel/SwfJsonLanguageService.ts [hunk 1]
```diff
@@ -0,0 +1,594 @@
...
+          getServices: () => Promise<SwfServiceCatalogService[]>;
+        };
+        relative: {
+          getServices: (textDocument: TextDocument) => Promise<SwfServiceCatalogService[]>;
+        };
+        getServiceFileNameFromSwfServiceCatalogServiceId: (swfServiceCatalogServiceId: string) => Promise<string>;
+      };
+      config: {
+        getServiceRegistryUrl: () => string | undefined; //FIXME: tiago make async
+        getServiceRegistryAuthInfo: () => { username: string; token: string } | undefined; //FIXME: tiago make async
```
Comment (tomasdavidorg) [inline packages/serverless-workflow-language-service/src/channel/SwfJsonLanguageService.ts]: Could we replace the FIXME and TODO with some jira links or can we remove them?

File: packages/vscode-extension-serverless-workflow-editor/src/extension/ServerlessWorkflowEditorChannelApiProducer.ts [hunk 2]
```diff
@@ -54,54 +55,16 @@ export class ServerlessWorkflowEditorChannelApiProducer implements KogitoEditorC
...
-      configuration: this.args.configuration,
-    });
-
-    const swfServiceCatalogStore = new SwfServiceCatalogStore({
-      fsWatchingServiceCatalogStore,
-      rhhccServiceRegistryServiceCatalogStore,
-    });
-
-    // TODO: This is a workaround
+    // TODO: This casting to `unknown` first is a workaround
```
Comment (tomasdavidorg) [inline packages/vscode-extension-serverless-workflow-editor/src/extension/ServerlessWorkflowEditorChannelApiProducer.ts:60]: Another TODO

File: packages/vscode-extension-serverless-workflow-editor/src/extension/languageService/VsCodeSwfLanguageService.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+            return swfServiceCatalogRelativeStore.getServices();
+          },
+        },
+        getServiceFileNameFromSwfServiceCatalogServiceId: async (swfServiceCatalogServiceId) => {
+          return getServiceFileNameFromSwfServiceCatalogServiceId(swfServiceCatalogServiceId);
+        },
+      },
+      config: {
+        shouldDisplayRhhccIntegration: async () => {
+          // FIXME: Tiago: This should take the OS into account as well. RHHCC integration only works on macOS.
```
Comment (tomasdavidorg) [inline packages/vscode-extension-serverless-workflow-editor/src/extension/languageService/VsCodeSwfLanguageService.ts]: Another FIXME

File: packages/vscode-extension/src/KogitoEditorChannelApiImpl.ts [hunk 4]
```diff
@@ -48,41 +46,71 @@ export class KogitoEditorChannelApiImpl implements KogitoEditorChannelApi, JavaC
...
+
+        this.editor.startListeningToDocumentChanges();
+        changeDocumentSubscription.dispose();
+      });
+
+      const { content } = await this.editor.envelopeServer.envelopeApi.requests.kogitoEditor_contentRequest();
+
+      const edit = new vscode.WorkspaceEdit();
+
+      // TODO: This shouldn't be a replace all the time. More conscious changes lead to better undo/redo stack.
```
Comment (tomasdavidorg) [inline packages/vscode-extension/src/KogitoEditorChannelApiImpl.ts:74]: Another TODO

**Comment by tiagobento** [comment]
@tomasdavidorg Thanks for the review!!

I just added the Jiras to the TODOs/FIXMEs and fixed the issue you found.

Now you should be able to use the extension normally even without the Red Hat Authentication extension installed.

**Review by tomasdavidorg — APPROVED** [review]
Thanks @tiagobento. Looks good to me now.

**Pull request merged**